scripts/pre_commit_linter.py

```diff
@@ -1612,262 +1362,1049 @@ def _check_mandatory_patterns(self):
+            self._match_line_breaks_in_controller_dependencies())
+
+        all_messages = (
+            common_messages + extra_js_files_messages +
+            js_and_ts_component_messages + directive_scope_messages +
+            sorted_dependencies_messages + controller_dependency_messages)
+        return all_messages
+
+    def _check_html_directive_name(self):
+        """This function checks that all HTML directives end
+        with _directive.html.
+        """
+        if self.verbose_mode_enabled:
+            print 'Starting HTML directive name check'
+            print '----------------------------------------'
+        total_files_checked = 0
+        total_error_count = 0
+        files_to_check = [
+            filepath for filepath in self.all_filepaths if not
+            any(fnmatch.fnmatch(filepath, pattern) for pattern in
+                EXCLUDED_PATHS)]
+        failed = False
+        summary_messages = []
+        # For RegExp explanation, please see https://regex101.com/r/gU7oT6/37.
+        pattern_to_match = (
+            r'templateUrl: UrlInterpolationService\.[A-z\(]+' +
+            r'(?P<directive_name>[^\)]+)')
+        with _redirect_stdout(_TARGET_STDOUT):
+            for filepath in files_to_check:
+                file_content = FileCache.read(filepath)
+                total_files_checked += 1
+                matched_patterns = re.findall(pattern_to_match, file_content)
+                for matched_pattern in matched_patterns:
+                    matched_pattern = matched_pattern.split()
+                    directive_filepath = ''.join(matched_pattern).replace(
+                        '\'', '').replace('+', '')
+                    if not directive_filepath.endswith('_directive.html'):
+                        failed = True
+                        total_error_count += 1
+                        print (
+                            '%s --> Please ensure that this file ends'
+                            'with _directive.html.' % directive_filepath)
+                        print ''
+
+            if failed:
+                summary_message = '%s   HTML directive name check failed' % (
+                    _MESSAGE_TYPE_FAILED)
+                summary_messages.append(summary_message)
+            else:
+                summary_message = '%s   HTML directive name check passed' % (
+                    _MESSAGE_TYPE_SUCCESS)
+                summary_messages.append(summary_message)
+
+            print ''
+            if total_files_checked == 0:
+                if self.verbose_mode_enabled:
+                    print 'There are no files to be checked.'
+            else:
+                print '(%s files checked, %s errors found)' % (
+                    total_files_checked, total_error_count)
+                print summary_message
+
+        return summary_messages
+
+
+class OtherLintChecksManager(LintChecksManager):
+    """Manages all the Other linting functions.
```

Comment:
Note here that we aren't checking JS/TS files. If possible, be explicit in what we are linting.

Reply:
Done